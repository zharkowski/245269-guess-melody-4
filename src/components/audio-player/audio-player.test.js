import React from "react";
import renderer from "react-test-renderer";
import AudioPlayer from "../audio-player/audio-player.jsx";

const mock = {
  song: {
    src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`
  }
};

describe(`Render AudioPlayer`, () => {
  it(`Should AudioPlayer render correctly`, () => {
    const {song} = mock;
    const tree = renderer.create(
        <AudioPlayer
          isPlaying={false}
          isLoading={true}
          src={song.src}
          onPlayButtonClick={() => {}}
        >
          <audio />
        </AudioPlayer>, {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
