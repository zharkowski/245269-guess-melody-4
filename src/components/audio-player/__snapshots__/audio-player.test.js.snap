// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render AudioPlayer Should AudioPlayer render correctly 1`] = `
<div
  className="track"
>
  <button
    className="track__button track__button--play"
    disabled={true}
    onClick={[Function]}
    type="button"
  />
  <div
    className="track__status"
  >
    <audio
      src="https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg"
    />
  </div>
</div>
`;
